Public/Src/Engine/Scheduler/Scheduler.cs

```diff
@@ -3512,6 +3525,9 @@ private async Task<PipExecutionStep> ExecutePipStep(RunnablePip runnablePip)
                         }
 
                         processRunnable.Executed = true;
+                        processRunnable.Process.SourceChangeAffectedOutputFiles = GetSourceChangeAffectedOutputs().ToReadOnlyArray();
+                        processRunnable.Process.SourceChangeAffectedOutputDirectories = GetSourceChangeAffectedOutputDirectories().ToReadOnlyArray();
+

```

Comment:
We should not update process pip. You can think of it as immutable structure once it is created.